install.sh

```diff
@@ -289,8 +289,27 @@ GASNET_CC="`$PKG_CONFIG $pkg --variable=GASNET_CC`"
 GASNET_CFLAGS="`$PKG_CONFIG $pkg --variable=GASNET_CFLAGS`"
 GASNET_CPPFLAGS="`$PKG_CONFIG $pkg --variable=GASNET_CPPFLAGS`"
 
-if [ "$GASNET_CC" != "$FPM_CC" ]; then 
-  echo "GASNET_CC=$GASNET_CC" and  "FPM_CC=$FPM_CC don't match"
+# Check whether GASNet was installed using Spack. If yes, bail out.
+# Note: relies on the fact that most Spack installations have "opt/spack"
+#       in the directory path, and assumes that the first directory returned
+#       by pkg-config contains the GASNet lib directory
+GASNET_LIBDIR="$(echo $GASNET_LIBS | awk '{print $1};')"
+if [ "$(echo $GASNET_LIBDIR | grep -q spack)" ]; then
```

Comment:
@wyphan let's use `sh` [built-in glob-matching] rather than `grep` to avoid forking

[built-in glob-matching]: https://stackoverflow.com/a/23472637